### PR TITLE
Minor enhancements

### PR DESCRIPTION
**Pull request by setrofim**
- Add missing __str__ implementations to some classes to improve
  readability of logs/interactive shell output.
- Add the commit hash (if available) to the version recorded in run.log.

### CODE REVIEW

File: wa/framework/version.py [hunk 1]
```diff
@@ -25,3 +27,22 @@ def get_wa_version():
...
     return version_string
+
+
+def get_wa_version_with_commit():
+    version_string = get_wa_version()
+    commit = get_commit()
+    if commit:
+        return '{}-{}'.format(version_string, commit)
+    else:
+        return commit
```
Comment (marcbonnici) [inline wa/framework/version.py]: This should be `version_string`

Comment (setrofim) [reply wa/framework/version.py]: Yes. Yes should.

Comment (setrofim) [reply wa/framework/version.py]: Fixed.

**Comment by setrofim** [comment]
Comment addressed.

**Pull request merged**